src/main/java/edu/cornell/kfs/module/ar/service/impl/CuCustomerAddressHelperServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants.ContractsAndGrantsInvoiceDocumentCreationProcessType;
+
+import edu.cornell.kfs.module.ar.service.CuCustomerAddressHelperService;
+
+public class CuCustomerAddressHelperServiceImpl implements CuCustomerAddressHelperService {
+    
+    private static final Logger LOG = LogManager.getLogger();
+    
+    /*
+     * CU Customization: KFSPTS-26393
+     * When creationProcessType is BATCH, validate that the customer number on the award as part if the customer
+     * address composite key matches the customer number associated to the agency associated to the award.
+     */
+    public boolean agencyCustomerMatchesAwardCustomer(ContractsAndGrantsBillingAward award,
+            ContractsAndGrantsInvoiceDocumentCreationProcessType creationProcessType) {
+        
+        if (ContractsAndGrantsInvoiceDocumentCreationProcessType.BATCH == creationProcessType) {
+            if (ObjectUtils.isNotNull(award.getAgency().getCustomer().getCustomerNumber()) &&
+                    ObjectUtils.isNotNull(award.getCustomerNumber()) &&
+                    StringUtils.isNotBlank(award.getAgency().getCustomer().getCustomerNumber()) &&
+                    StringUtils.isNotBlank(award.getCustomerNumber()) &&
+                    award.getAgency().getCustomer().getCustomerNumber().equalsIgnoreCase(award.getCustomerNumber())) {
+                return true;
+            } else {
+                LOG.info("agencyCustomerMatchesAwardCustomer: Mismatched Customer records detected for " +
+                         "Award.Agency and Award.CustomerAddress : Award.Agency.CustomerNumber = " +
+                            (ObjectUtils.isNull(award.getAgency()) ? "null" : (ObjectUtils.isNull(award.getAgency().getCustomer().getCustomerNumber()) ? "null" : award.getAgency().getCustomer().getCustomerNumber())) +
+                        " Award.CustomerNumber = " + award.getCustomerNumber() +
+                        " Award.CustomerAddressIdentifier = " +
+                         (ObjectUtils.isNull(award.getCustomerAddressIdentifier()) ? "null" : award.getCustomerAddressIdentifier().intValue()));
+                return false;
+            } 
+        } else {
+            LOG.info("agencyCustomerMatchesAwardCustomer: Validation bypassed due to creationProcessType being " + creationProcessType.getName());
+            return true;
+        }
+    }
+    
+    /*
+     * CU Customization: KFSPTS-26393
+     * When creationProcessType is BATCH, validate that the customer address identifier on the award exists.
+     */
+    public boolean customerAddressIdentifierExists(ContractsAndGrantsBillingAward award,
+            ContractsAndGrantsInvoiceDocumentCreationProcessType creationProcessType) {
+        
+        if (ContractsAndGrantsInvoiceDocumentCreationProcessType.BATCH == creationProcessType) {
+            if (ObjectUtils.isNotNull(award.getCustomerAddressIdentifier())) {
```

Comment:
Is this only supposed to be a null check? Or is it supposed to do a check against the database?

Reply:
My understanding after talking with the functional user was to report validation errors in a way that they could correct the data. Later downstream CGB CINV processing performs the existence check for the address. This check was purposely code to not be a database check (just prevent NPEs) so we were not incurring more processing overhead.

Reply:
This has now been updated to also check against the DB to see if the address exists.